lib/authorizer/edgegrid.js

```diff
@@ -0,0 +1,341 @@
+        return callback();
+    };
+
+/**
+ * @implements {AuthHandlerInterface}
+ */
+module.exports = {
+    /**
+     * @property {AuthHandlerInterface~AuthManifest}
+     */
+    manifest: {
+        info: {
+            name: 'edgegrid',
+            version: '1.0.0'
+        },
+        updates: [
+            {
+                property: 'Authorization',
+                type: 'header'
+            }
+        ]
+    },
+
+    /**
+     * Initializes a item (fetches all required parameters, etc) before the actual authorization step.
+     *
+     * @param {AuthInterface} auth
+     * @param {Response} response
+     * @param {AuthHandlerInterface~authInitHookCallback} done
+     */
+    init: function (auth, response, done) {
+        done(null);
+    },
+
+    /**
+     * Checks the item, and fetches any parameters that are not already provided.
+     *
+     * @param {AuthInterface} auth
+     * @param {AuthHandlerInterface~authPreHookCallback} done
+     */
+    pre: function (auth, done) {
+        // only check required auth params here
+        done(null, Boolean(auth.get('accessToken') && auth.get('clientToken') && auth.get('clientSecret')));
+    },
+
+    /**
+     * Verifies whether the request was successful after being sent.
+     *
+     * @param {AuthInterface} auth
+     * @param {Requester} response
+     * @param {AuthHandlerInterface~authPostHookCallback} done
+     */
+    post: function (auth, response, done) {
+        done(null, true);
+    },
+
+    /**
+     * Generates the signature, and returns the Authorization header.
+     *
+     * @param {Object} params
+     * @param {String} params.accessToken Access token provided by service provider
+     * @param {String} params.clientToken Client token provided by service provider
+     * @param {String} params.clientSecret Client secret provided by service provider
+     * @param {String} params.nonce Nonce to include in authorization header
+     * @param {String} params.timestamp Timestamp as defined in protocol specification
+     * @param {String} [params.bodyHash] Base64-encoded SHA–256 hash of request body for POST request
+     * @param {Object[]} params.headers Request headers
+     * @param {String[]} params.headersToSign Ordered list of headers to include in signature
+     * @param {String} params.method Request method
+     * @param {Url} params.url Node's URL object
+     * @returns {String} Authorization header
+     */
+    computeHeader: function (params) {
+        var authHeader = 'EG1-HMAC-SHA256 ',
+            signingKey = base64HmacSha256(params.timestamp, params.clientSecret),
+            dataToSign;
+
+        authHeader += `client_token=${params.clientToken};`;
+        authHeader += `access_token=${params.accessToken};`;
+        authHeader += `timestamp=${params.timestamp};`;
+        authHeader += `nonce=${params.nonce};`;
+
+        dataToSign = [
+            params.method,
+            params.url.protocol.replace(COLON, EMPTY),
```

Comment:
3 level nested 😨 
make sure the necessary checks for `params`  was done before.